src/CodeGen_Hexagon.cpp

```diff
@@ -1297,8 +1312,17 @@ void CodeGen_Hexagon::visit(const Div *op) {
 }
 
 void CodeGen_Hexagon::visit(const Cast *op) {
-    // TODO: Do we need to handle same-sized vector casts before LLVM sees them?
-    CodeGen_Posix::visit(op);
+    if (op->value.type().bits() == 1 && op->type.bits() > 1) {
+        // If left to LLVM, this maps true to one. However,
+        // eliminate_bool_vectors assumes that it maps to negative
+        // one. This will map true to negative one instead.
+        value = call_intrin(op->type,
```

Comment:
I'm not talking about relational operators -- I'm talking about upcasting a const bool. Cast(Uint(8), x < y) returning 255 instead of 1 seems like a bug in our OpenCL backend.